### PR TITLE
feat(examples): Introduce Caddy as example

### PR DESCRIPTION
**Pull request by razvand**
Add Caddy as binary compatibility run. Build Caddy as static PIE application using `Dockerfile`. Then run it with the `base` image from the registry.

Add

* `Dockerfile`: filesystem, including binary and libraries
* `Makefile`: used to build the filesystem / export the from the `Dockerfile`
* `Kraftfile`: build / run rules, including pulling the `base` image
* `README.md`: document how to use
* `hosts`, `Caddyfile`: required Caddy configuration files

### CODE REVIEW

**Review by StefanJum — APPROVED** [review]
All good, thanks.

Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Review by Mihnea0Firoiu — COMMENTED** [review]
I don't know if it is supposed to happen, but running **./kraft-run-qemu.sh** gives the output:
```
 E  machine(s) not found
 E  machine(s) not found
virbr0
 E  coud not find runtime 'index.unikraft.io/unikraft.org/base'
```
**kraft version:** kraft 0.7.0 (b4c61f166ade831a725e7c16edebc810f7c4329a) go1.21.2 2023-11-08T02:09:50Z

**Comment by razvand** [comment]
> I don't know if it is supposed to happen, but running **./kraft-run-qemu.sh** gives the output:
> 
> ```
>  E  machine(s) not found
>  E  machine(s) not found
> virbr0
>  E  coud not find runtime 'index.unikraft.io/unikraft.org/base'
> ```
> 
> **kraft version:** kraft 0.7.0 (b4c61f166ade831a725e7c16edebc810f7c4329a) go1.21.2 2023-11-08T02:09:50Z

@Mihnea0Firoiu , it was an issue on my side. I fixed it.

**Review by StefanJum — CHANGES_REQUESTED** [review]
The qemu scripts work fine, the `kraft` commands/scripts show the same `could not find runtime: too many options` that I get on #5.

Also, is this not supposed to be used with the `utils/bincompat` scripts from #8 ? There is no `config.yaml` file and the scripts are in the PR as well.

File: examples/caddy-2.7/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+1. [Install Unikraft's companion command-line toolchain `kraft`](https://unikraft.org/docs/cli).
+
+1. Install Docker following the [official instructions](https://docs.docker.com/engine/install/).
+   This can be either [Docker Engine](https://docs.docker.com/engine/) or [Docker Desktop](https://docs.docker.com/desktop/).
+
+1. Build the Caddy root filesystem from the `Dockerfile` using:
+
+   ```console
+   make export
```
[the remark this answers is not in the document]
Comment (razvand) [reply examples/caddy-2.7/README.md:16]: No reason. Will fix.

**Review by Mihnea0Firoiu — COMMENTED** [review]
The scripts work as intended, however the ```kraft``` command gives the output: ```E  could not find runtime: too many options```

**Review by StefanJum — CHANGES_REQUESTED** [review]
One small comment.

Besides that, I would add the `protocol not available` error in the README, just so people don't get scared by it and assume something failed.

**Pull request closed without merging**